query-grammar/src/query_grammar.rs

```diff
@@ -1283,6 +1283,10 @@ mod test {
             super::field_name("~my~field:a"),
             Ok(("a", "~my~field".to_string()))
         );
+        assert_eq!(
+            super::field_name(".my.field.name : a"),
```

Comment:
can you also add a `test_parse_query_to_ast_helper` test somewhere to make sure this works (and stays working) for both the strict and the lenient parser